src/components/LoanCards/GridLoanCard.vue

```diff
@@ -196,24 +197,36 @@ export default {
 	methods: {
 		toggleFavorite() {
 			// optimistically toggle it locally first
-			this.isFavorite = !this.isFavorite;
+			// this.isFavorite = !this.isFavorite;
 
 			this.apollo.mutate({
 				mutation: loanFavoriteMutation,
 				variables: {
 					loan_id: this.loan.id,
 					is_favorite: this.isFavorite
 				}
-			}).then(({ data }) => {
-				if (data) {
-					// @todo - provide a better soft-landing if mutation failed
-					const favorite = _get(data, 'loan.favorite');
-
-					if (favorite === null) {
-						this.isFavorite = !this.isFavorite;
+			}).then(data => {
+				if (data.errors) {
+					_forEach(data.errors, ({ message }) => {
+						this.$showTipMsg(message, 'error');
+					});
+				} else {
```

Comment:
You probably want to also set this.isFavorite just inside this else statement using the return value from your mutation "data".